pkg/securitypolicy/framework.rego

```diff
@@ -393,6 +526,14 @@ overlay_matches {
     layerPaths_ok(container.layers)
 }
 
+overlay_matches {
+    some issuer in data.metadata.issuers
+    some feed in issuer.feeds
+    some fragment in feed
+    some container in feed.containers
```

Comment:
Shouldn't this be "some container in fragment.containers"

Reply:
do we need information about what issuer/feed/fragment matched a rule? If not, can we make this a comprehension function and reuse? Or maybe there's something more memory efficient like a generators in python?

Reply:
Great catch @hgarvison! This is a tricky thing that gets lost in our tests because it produces the expected error message but for the wrong reasons.

@anmaxvl In Rego you can definitely create a comprehension but it will actually be much less performant. Doing it this way means that once anything is found which fulfills the condition then rule evaluation halts (i.e. early stopping). If you do a comprehension then it will find all of the items which match the condition, which adds needlessly computational complexity.

Reply:
@matajoh no worries, that's what code reviews are for